### PR TITLE
Add CLI Tests

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
## Description

Provide a summary of the modification as well as the issue that has been resolved.

- This is missing Predict and export tests

## Changes

<details>
<summary>Describe the changes you made</summary>

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which refactors the code base)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

</details>

## Checklist

<details>
<summary>Ensure that you followed the following</summary>

- [ ] I have added a summary of my changes to the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) (not for minor changes, docs and tests).
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas (If applicable)
- [ ] I have made corresponding changes to the documentation (If applicable)
- [ ] I have added tests that prove my fix is effective or that my feature works (If applicable)

</details>

### CODE REVIEW

File: tests/pre_merge/v1/__init__.py
Comment (samet-akcay) [inline tests/pre_merge/v1/__init__.py:1]: Would it be possible to organize the the tree structure of the `tests` as follows:

```bash
└── v1
    ├── conftest.py
    ├── helpers
    ├── __init__.py
    ├── integration
    ├── regression
    └── unit
```

We could then remove `v1`, and these subdirectories would be within `tests` directory. 

I think we don't need to have `pre_merge` directory. We could have a `pre_merge` in `tox`, which could run `unit` and `integration` tests.

File: src/anomalib/data/ucsd_ped.py [hunk 1]
```diff
@@ -169,8 +169,8 @@ def __init__(
         category: str,
         transform: A.Compose,
         split: Split,
-        clip_length_in_frames: int = 1,
-        frames_between_clips: int = 1,
+        clip_length_in_frames: int = 2,
+        frames_between_clips: int = 10,
```
Comment (samet-akcay) [inline src/anomalib/data/ucsd_ped.py:173]: Can you explain this change?

Comment (ashwinvaidya17) [reply src/anomalib/data/ucsd_ped.py:173]: AI Vad uses these parameters so I decided to keep these as the default value for USCD. This way people can use `anomalib fit --model anomalib.models.AIVad --data anomalib.data.USCD `

File: src/anomalib/engine/engine.py [hunk 2]
```diff
@@ -93,11 +92,21 @@ class Engine:
     """Anomalib Engine.
 
     Note:
-    ----
         Refer to PyTorch Lightning's Trainer for a list of parameters for details on other Trainer parameters.
 
     Args:
-        callbacks: Add a callback or list of callbacks.
+        callbacks (list[Callback]): Add a callback or list of callbacks.
```
Comment (samet-akcay) [inline src/anomalib/engine/engine.py:98]: if this is a callback or list of callbacks, then is the annotation correct?

File: src/anomalib/data/ucsd_ped.py (hunk 1 above)
Comment (samet-akcay) [reply src/anomalib/data/ucsd_ped.py:173]: thanks, it is clear now

**Pull request merged**